File: objc/ChainOfResponsibility/ChainOfResponsibility/GeneralDivison.m
#import "GeneralDivison.h"

@implementation GeneralDivison

-(void)handle:(Request*)request {
    if(request.weight <= 10 && request.money <= 1000)
        [self processRequest];
    else [super handle:request];
}

-(void)processRequest {
    NSLog(@"This request is processed by -- General Divison");
}

@end

File: swift/ChainOfResponsibility/ChainOfResponsibility/GeneralDivison.swift
import Foundation

public class GeneralDivison : Divison {
    
    override func handle(_ request: Request) {
        if request.weight <= 10 && request.money <= 1000 {
            processRequest()
        } else {
            super.handle(request)
        }
    }
    
    override func processRequest() {
        print("This request is processed by -- General Divison")
    }
}
